vmware-tanzu/velero
# fix: Typos and add more spell checking rules to CI

Thank you for contributing to Velero!

# Please add a summary of your change

Fix some typos in the project and add spell checking rules to enforce [Velero.io word list](https://velero.io/docs/v1.11/style-guide/#veleroio-word-list).

Most of the typos I used [CSpell](https://github.com/streetsidesoftware/cspell) to catch them, and some I used [copespell](https://github.com/codespell-project/codespell) itself (which is used by the the project CI), but with one more flag: `--builtin`.

To try out these approaches, you can create Dockerfiles with each technology (but there will be a lot of false positives).

## To test CSpell

create `cspell.yaml` and `Dockerfile.cspell` in the root of the project, with the following contents
```yaml
version": "0.2"
language: "en"
caseSensitive: true
words: [
  "Kubernetes",
  "Docker",
  "Velero",
  "VMware",
  "Heptio",
  "restic",
  "kopia",
]
patterns:
  - name: github-usernames
    pattern: /@[a-zA-Z\d](?:[a-zA-Z\d]|-(?=[a-zA-Z\d])){0,38}/g
ignoreRegExpList:
  - github-usernames
ignorePaths: [
  ".git",
  "*.png",
  "*.jpg",
  "*.eps",
  "*.svg",
  "*.gif",
  "*.ico",
  "*.woff",
  "*.ttf",
  "./config/crd/v1beta1/crds/crds.go",
  "./config/crd/v1/crds/crds.go",
  "./config/crd/v2alpha1/crds/crds.go",
  "./go.sum",
  "./LICENSE",
  "site/content/docs/v1.10",
  "site/content/docs/v1.9",
  "site/content/docs/v1.8",
  "site/content/docs/v1.7",
  "site/content/docs/v1.6",
  "site/content/docs/v1.5",
  "site/content/docs/v1.4",
  "site/content/docs/v1.3.2",
  "site/content/docs/v1.3.1",
  "site/content/docs/v1.3.0",
  "site/content/docs/v1.2.0",
  "site/content/docs/v1.1.0",
  "site/content/docs/v1.0.0",
  "site/content/docs/v0.11.0",
  "site/content/docs/v0.10.0",
  "site/content/docs/v0.9.0",
  "site/content/docs/v0.8.1",
  "site/content/docs/v0.8.0",
  "site/content/docs/v0.7.1",
  "site/content/docs/v0.7.0",
  "site/content/docs/v0.6.0",
  "site/content/docs/v0.5.0",
  "site/content/docs/v0.4.0",
  "site/content/docs/v0.3.0",
]
```
```dockerfile
FROM node:16.16.0-alpine

WORKDIR /velero/

RUN npm install -g cspell@latest

CMD cspell "**" \
    --gitignore \
    --dot \
    --show-suggestions
```
and run
```
docker image build --tag cspell - < Dockerfile.cspell
docker container run --rm -it -v "./:/velero" cspell
``` 
You will get something like `CSpell: Files checked: 2380, Issues found: 34701 in 2141 files`.

## To test codespell

create `Dockerfile.codespell` in the root of the project, with the following content
```dockerfile
FROM python:3.11.4-slim-bullseye

WORKDIR /velero/

RUN pip install codespell[toml]>=2.2.5

CMD codespell \
    --check-filenames \
    --check-hidden \
    --skip .git,*.png,*.jpg,*.woff,*.ttf,*.gif,*.ico,./config/crd/v1beta1/crds/crds.go,./config/crd/v1/crds/crds.go,,./config/crd/v2alpha1/crds/crds.go,./go.sum,./LICENSE \
    --ignore-words-list iam,aks,ist,bridget,ue,shouldnot,atleast \
    --count \
    --summary \
    --builtin "clear,rare,informal,usage,code,names"
```
and run
```
docker image build --tag codespell - < Dockerfile.codespell
docker container run --rm -it -v "./:/velero" codespell
```

# Does your change fix a particular issue?

No.

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x] Updated the corresponding documentation in `site/content/docs/main`.
